leaderelection/src/main/java/io/scalecube/leaderelection/HeartbeatScheduler.java

```diff
@@ -0,0 +1,62 @@
+package io.scalecube.leaderelection;
+
+import io.scalecube.cluster.ICluster;
+import io.scalecube.transport.Message;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * heartbeat scheduler broadcast heartbeats on the cluster to maintain leadership, members in the cluster expects this
+ * heartbeat. in case heartbeat does not reach members within X time leader election process is initiated.
+ * 
+ * @author Ronen Nachmias
+ */
+
+public class HeartbeatScheduler {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(HeartbeatScheduler.class);
+  private final ScheduledExecutorService scheduler;
+  private final ICluster cluster;
+  private final long heartbeatInterval;
+
+  public HeartbeatScheduler(ICluster cluster, long heartbeatInterval) {
+    this.cluster = cluster;
+    this.heartbeatInterval = heartbeatInterval;
+    String nameFormat = "sc-leaderelection-heartbeat" + cluster.address().toString();
+    scheduler =Executors.newSingleThreadScheduledExecutor(
+        new ThreadFactoryBuilder().setNameFormat(nameFormat).setDaemon(true).build());
+  }
+
+  /**
+   * when becoming a leader the leader schedule heatbeats and maintain leadership followers are expecting this heartbeat
+   * in case the heartbeat will not arrive in X time they will assume no leader.
+   */
+  public void schedule() {
```

Comment:
It's pretty hard to read this method's javadoc and also javadoc comment of this class, please fix typos, add punctuation marks, start new sentence from capital letter and so.